File: compute_psnr.py
from scipy.misc import imread,imsave
import numpy as np

def ycbcr2rgb(im):
    xform = np.array([[1, 0, 1.402], [1, -0.34414, -.71414], [1, 1.772, 0]])
    rgb = im.astype(np.float)
    rgb[:,:,[1,2]] -= 128
    return rgb.dot(xform.T)

gt = imread('butterfly_GT.bmp')
gt = ycbcr2rgb(gt)
gt = gt[:,:,0]
gt = gt[6:231,6:231,:]
imsave('ground_true_Y.png', gt)

test = imread('sr.bmp')
test = ycbcr2rgb(test)
test = test[:,:,0]
#imsave('test.png', test)

File: computepsnr.m
gt = imread('butterfly_GT.bmp');
gt = rgb2ycbcr(gt);
gt = gt(:,:,1);
gt = gt(7:231,7:231,1);
%gt = gt(7:255,7:255,1);

psnrs = [];
for i = 3200:50:14000
    fn = sprintf('outputs/%05d.png',i);
    ti = imread(fn);
    ti = ti(7:231,7:231,:);
    if size(ti,3) > 1
        ti = rgb2ycbcr(ti);
        ti = ti(:,:,1);
    end
    psnrs = [psnrs psnr(ti, gt)];
end